### DISCUSSION TITLE
How to use url slug instead of id

### DISCUSSION THREAD

**Discussion by khuongviettai** [vercel/next.js#66932] [App Router] [answered]
I want to create a website but I don't know how to use the url according to the product name instead of the product id. How can I fetch the exact product I choose?

for example, i want the url to be: /products/t-shirt instead of /products/123

**Comment by icyJoseph** [comment]
The API you use to get products from you database, or other source of truth, has to be able to do a `getProductBySlug` kind of search. And it should also give you a list of product slugs, to map over to generate, links and pages.

**Reply by khuongviettai** [reply]
What if two products have the same name? 
For example: t-shirt has id 1 and 1 t-shirt has id 2. Now we have a link like this ```products/t-shirt```. I want to get id 2, how can I do that?

**Reply by icyJoseph** [reply, marked as answer]
Then you should've designed it, so that they are unique 😅 

You could also maybe do `t-shirt---id` or something like that, and then split on `---` to collect the `id`. In general, append something to the end of the URL, which you can use to figure out which product to show. Keep them short though.

**Reply by khuongviettai** [reply]
nice
